torchgeo/datamodules/spacenet.py

```diff
@@ -0,0 +1,183 @@
+    .. versionadded:: 0.4
+    """
+
+    def __init__(
+        self,
+        batch_size: int = 64,
+        num_workers: int = 0,
+        val_split_pct: float = 0.1,
+        test_split_pct: float = 0.2,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a LightningDataModule for SpaceNet1.
+
+        Args:
+            batch_size: The batch size to use in all created DataLoaders
+            num_workers: The number of workers to use in all created DataLoaders
+            val_split_pct: What percentage of the dataset to use as a validation set
+            test_split_pct: What percentage of the dataset to use as a test set
+            **kwargs: Additional keyword arguments passed to
+                :class:`~torchgeo.datasets.SpaceNet1`
+        """
+        super().__init__()
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.val_split_pct = val_split_pct
+        self.test_split_pct = test_split_pct
+        self.kwargs = kwargs
+
+        self.padto = K.PadTo((448, 448))
+
+    def on_after_batch_transfer(
+        self, batch: Dict[str, Any], batch_idx: int
+    ) -> Dict[str, Any]:
+        """Apply batch augmentations after batch is transferred to the device.
+
+        Args:
+            batch: mini-batch of data
+            batch_idx: batch index
+
+        Returns:
+            augmented mini-batch
+        """
+        if (
+            hasattr(self, "trainer")
+            and self.trainer is not None
+            and hasattr(self.trainer, "training")
+            and self.trainer.training
+        ):
+            # Kornia expects masks to be floats with a channel dimension
+            x = batch["image"]
+            y = batch["mask"].float().unsqueeze(1)
+
+            train_augmentations = K.AugmentationSequential(
+                K.RandomRotation(p=0.5, degrees=90),
+                K.RandomHorizontalFlip(p=0.5),
+                K.RandomVerticalFlip(p=0.5),
+                K.RandomSharpness(p=0.5),
+                K.ColorJitter(
+                    p=0.5,
+                    brightness=0.1,
+                    contrast=0.1,
+                    saturation=0.1,
+                    hue=0.1,
+                    silence_instantiation_warning=True,
+                ),
+                data_keys=["input", "mask"],
+            )
+            x, y = train_augmentations(x, y)
+
+            # torchmetrics expects masks to be longs without a channel dimension
+            batch["image"] = x
+            batch["mask"] = y.squeeze(1).long()
+
+        return batch
+
+    def preprocess(self, sample: Dict[str, Any]) -> Dict[str, Any]:
+        """Transform a single sample from the Dataset.
+
+        Args:
+            sample: dictionary containing image and mask
+
+        Returns:
+            preprocessed sample
+        """
+        sample["image"] = sample["image"].float()
+        sample["image"] = self.padto(sample["image"]).squeeze()
+        sample["image"] /= 255.0
```

Comment:
```suggestion
        sample["image"] = sample["image"].float() / 255
        sample["image"] = self.padto(sample["image"]).squeeze()
```
Division before padding is faster.